src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java

```diff
@@ -292,20 +293,36 @@ public void checkBundleAndSharedUser(AthenaUser user, DownloadBundle bundle){
     @Override
     public List<License> saveLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
 
-        Iterable<License> licenses = vocabularyV4Ids.stream()
-                .map(vocabularyV4Id -> new License(user, new VocabularyConversion(vocabularyV4Id), status))
-                .collect(toList());
-        List<License> result = new ArrayList<>();
-        licenseRepository.save(licenses).iterator().forEachRemaining(result::add);
+        final List<License> newLicenses = buildLicenses(user, vocabularyV4Ids, status);
+        final List<License> savedLicenses = licenseRepository.save(newLicenses);
 
         conceptService.invalidateGraphCache(user.getId());
-        return result;
+        return savedLicenses;
+    }
+
+    @Override
+    public List<License> saveLicenses(AthenaUser user, List<License> licenses) {
+
+        final List<License> savedLicenses = licenseRepository.save(licenses);
+
+        conceptService.invalidateGraphCache(user.getId());
+        return savedLicenses;
+    }
+
+    private List<License> buildLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
+
+        return vocabularyV4Ids.stream()
+                .map(VocabularyConversion::new)
+                .map(conversion -> new License(user, conversion, status))
+                .collect(toList());
     }
 
     @Override
     public Long requestLicenses(AthenaUser user, Integer vocabularyV4Id) {
 
-        return saveLicenses(user, Collections.singletonList(vocabularyV4Id), PENDING).get(0).getId();
+        final List<License> requestedLicenses = buildLicenses(user, singletonList(vocabularyV4Id), PENDING);
+        requestedLicenses.forEach(license -> license.setRequestDate(new Date()));
+        return saveLicenses(user, requestedLicenses).get(0).getId();
```

Comment:
this looks weird.  I would refactor this way: 
```
    @Override
    public Long requestLicense(AthenaUser user, Integer vocabularyV4Id) {

        final License license = buildLicense(user, vocabularyV4Id, PENDING);
        license.setRequestDate(new Date());

        final License savedLicense = licenseRepository.save(license);
        conceptService.invalidateGraphCache(user.getId());

        return savedLicense.getId();
    }

```

```
    private List<License> buildLicenses(AthenaUser user, List<Integer> vocabularyV4Ids, LicenseStatus status) {
        return vocabularyV4Ids.stream()
                .map(id -> buildLicense(user, id, status))
                .collect(toList());
    }

    private License buildLicense(AthenaUser user, Integer vocabularyV4Id, LicenseStatus status) {
        VocabularyConversion vocabularyConversion = new VocabularyConversion();
        return new License(user, vocabularyConversion, status)
    }
```

Reply:
I've made it shorter and cleaner instead.